### PR TITLE
Before writing to partition check for sufficient space on partition

### PR DESCRIPTION
**Pull request by zedi-pramodh**
1. Get the size of the image using contenttree status
2. Get the size of the partition using zboot
3. If partition size is < image size, error out
4. Fixed zboot API to be more portable

### CODE REVIEW

**Comment by zedi-pramodh** [comment]
Tried to upgrade to image of size 321MB

dev.log.2224562291:{"severity":"info","source":"zedagent","iid":"1865","content":"{\"error\":\"doBaseOsActivate: Image size 336605184 bytes greater than partition size 314572800 bytes\",\"error-time\":\"2023-06-10T03:02:30.384830099Z\",\"file\":\"/pillar/types/zedagenttypes.go:136\",\"func\":\"github.com/lf-edge/eve/pkg/pillar/types.BaseOsStatus.LogModify\",\"level\":\"info\",\"log_event_type\":\"log\",\"msg\":\"BaseOs status modify\",\"obj_key\":\"baseos_status-v7.4.0-master-private-1212-g92de24282-kubevirt-amd64\",\"obj_name\":\"v7.4.0-master-private-1212-g92de24282-kubevirt-amd64\",\"obj_type\":\"baseos_status\",\"pid\":1865,\"source\":\"zedagent\",\"state\":\"LOADED\",\"time\":\"2023-06-10T03:02:30.386236254Z\"}\n","msgid":3412,"timestamp":{"seconds":1686366150,"nanos":386236254},"filename":"/pillar/types/zedagenttypes.go:136","function":"github.com/lf-edge/eve/pkg/pillar/types.BaseOsStatus.LogModify"}


On the controller
=================
ALERTS
Date & Time                    Severity        Description
Jun 09, 2023  8:02 PM   Error            doBaseOsActivate: Image size 336605184 bytes greater than partition size 314572800 bytes

File: pkg/pillar/cmd/baseosmgr/handlebaseos.go [hunk 1]
```diff
@@ -259,6 +259,33 @@ func doBaseOsActivate(ctx *baseOsMgrContext, uuidStr string,
...
+		errString := fmt.Sprintf("doBaseOsActivate: ContentTreeStatus not found for %s", status.ContentTreeUUID)
+		log.Error(errString)
+		status.SetErrorNow(errString)
+		changed = true
+		return changed
+	}
+
+	if cts.State == types.LOADED {
+		partSize := zboot.GetPartitionSizeInBytes(status.PartitionLabel)
+		imageSize := cts.MaxDownloadSize
```
Comment (eriknordmark) [inline pkg/pillar/cmd/baseosmgr/handlebaseos.go:279]: Don't we have an actual size which could be smaller than MaxDownloadSize?

File: pkg/pillar/zboot/zboot.go [hunk 2]
```diff
@@ -256,11 +256,20 @@ func GetPartitionSizeInBytes(partName string) uint64 {
 	validatePartitionName(partName)
 	_, ok := partDev[partName]
 	if ok {
-		partsize, err := execWithRetry(nil, "zboot", "partdevsize", partName)
+		ret, err := execWithRetry(nil, "zboot", "partdevsize", partName)
 		if err != nil {
 			logrus.Fatalf("zboot partdevsize %s: err %v\n", partName, err)
```
Comment (eriknordmark) [inline pkg/pillar/zboot/zboot.go:261]: No need for \n at the end of the log strings; log is in json.

File: pkg/pillar/zboot/zboot.go [hunk 3]
```diff
@@ -256,11 +256,20 @@ func GetPartitionSizeInBytes(partName string) uint64 {
...
 		}
-		return binary.BigEndian.Uint64(partsize)
+		partsize := string(ret)
+		partsize = strings.TrimSpace(partsize)
+		logrus.Infof("partsize in bytes %s", partsize)
+		value, err := strconv.ParseUint(partsize, 10, 64)
+		if err != nil {
+			logrus.Errorf("Strconv failed for partsize err %v\n", err)
```
Comment (eriknordmark) [inline pkg/pillar/zboot/zboot.go:268]: Ditto

File: pkg/pillar/cmd/baseosmgr/handlebaseos.go (hunk 1 above)
Comment (zedi-pramodh) [reply pkg/pillar/cmd/baseosmgr/handlebaseos.go:279]: MaxDownloadSize seems to be the size of image sent from the controller in the content tree config. zedagent is parsing that value from the config. To me that seems to be the actual size of the image. Let me know if that is not correct.

File: pkg/pillar/zboot/zboot.go (hunk 2 above)
Comment (zedi-pramodh) [reply pkg/pillar/zboot/zboot.go:261]: ok

File: pkg/pillar/zboot/zboot.go (hunk 3 above)
Comment (zedi-pramodh) [reply pkg/pillar/zboot/zboot.go:268]: ok

**Review by eriknordmark — APPROVED** [review]
Kick off tests

**Pull request merged**